winml/test/adapter/AdapterSessionTest.cpp

```diff
@@ -0,0 +1,311 @@
+}
+
+void RegisterCustomRegistry() {
+  IMLOperatorRegistry* registry;
+  THROW_IF_NOT_OK_MSG(winml_adapter_api->CreateCustomRegistry(&registry), ort_api);
+  if (registry) {
+    const auto session_options = CreateUniqueOrtSessionOptions();
+    auto session = CreateUniqueOrtSession(session_options);
+    THROW_IF_NOT_OK_MSG(winml_adapter_api->SessionRegisterCustomRegistry(session.get(), registry), ort_api);
+  }
+}
+
+void RegisterCustomRegistry_DML() {
+  GPUTEST
+  IMLOperatorRegistry* registry;
+  THROW_IF_NOT_OK_MSG(winml_adapter_api->CreateCustomRegistry(&registry), ort_api);
+  WINML_EXPECT_NOT_EQUAL(nullptr, registry);
+  const auto session_options = CreateUniqueOrtSessionOptions();
+  auto session = CreateUniqueOrtSession(session_options);
+  THROW_IF_NOT_OK_MSG(winml_adapter_api->SessionRegisterCustomRegistry(session.get(), registry), ort_api);
+}
+
+void LoadAndPurloinModel(const UniqueOrtSession& session, const std::string& model_path) {
+  com_ptr<WinML::IModel> model;
+  WINML_THROW_IF_FAILED(engine_factory->CreateModel(model_path.c_str(), sizeof(model_path), model.put()));
+
+  Microsoft::WRL::ComPtr<WinML::IOnnxruntimeModel> onnxruntime_model;
+  WINML_EXPECT_HRESULT_SUCCEEDED(model->QueryInterface(IID_PPV_ARGS(&onnxruntime_model)));
+  OrtModel* ort_model;
+  WINML_EXPECT_HRESULT_SUCCEEDED(onnxruntime_model->DetachOrtModel(&ort_model));
+  THROW_IF_NOT_OK_MSG(winml_adapter_api->SessionLoadAndPurloinModel(session.get(), ort_model), ort_api);
+}
+
+void LoadAndPurloinModel() {
+  const auto session_options = CreateUniqueOrtSessionOptions();
+  auto session = CreateUniqueOrtSession(session_options);
+  LoadAndPurloinModel(session, "fns-candy.onnx");
+}
+
+void Initialize() {
+  const auto session_options = CreateUniqueOrtSessionOptions();
+  auto session = CreateUniqueOrtSession(session_options);
+
+  com_ptr<WinML::IModel> model;
+  const auto model_path = "fns-candy.onnx";
+  WINML_THROW_IF_FAILED(engine_factory->CreateModel(model_path, sizeof(model_path), model.put()));
+
+  Microsoft::WRL::ComPtr<WinML::IOnnxruntimeModel> onnxruntime_model;
+  WINML_EXPECT_HRESULT_SUCCEEDED(model->QueryInterface(IID_PPV_ARGS(&onnxruntime_model)));
+  OrtModel* ort_model;
+  WINML_EXPECT_HRESULT_SUCCEEDED(onnxruntime_model->DetachOrtModel(&ort_model));
+  THROW_IF_NOT_OK_MSG(winml_adapter_api->SessionLoadAndPurloinModel(session.get(), ort_model), ort_api);
+
+  THROW_IF_NOT_OK_MSG(winml_adapter_api->SessionInitialize(session.get()), ort_api);
+}
+
+class MockSink : public onnxruntime::logging::ISink {
+public:
+  static bool called;
+
+private:
+  void SendImpl(const onnxruntime::logging::Timestamp& timestamp, const std::string& logger_id, const onnxruntime::logging::Capture& message) override {
+    called = true;
+  }
+
+  void SendProfileEvent(onnxruntime::profiling::EventRecord&) const override { };
+};
+bool MockSink::called = false;
+
+void Profiling() {
+  ort_env->SetLoggingManager(nullptr);
```

Comment:
This is not a thread safe api to call, and it not exposed on the ABI. It is only there so that the WinML API can configure a sink and needs to be called using the same pattern in the ort lib.